src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobRejecter.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.sas.BlobContainerSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static uk.gov.hmcts.reform.blobrouter.services.storage.RejectedFilesHandler.REJECTED_CONTAINER_SUFFIX;
+
+@Component
+public class BlobRejecter {
+
+    private static final Logger logger = getLogger(BlobRejecter.class);
+
+    private final BlobServiceClient storageClient;
+
+    public BlobRejecter(BlobServiceClient storageClient) {
+        this.storageClient = storageClient;
+    }
+
+    public void moveToRejectedContainer(String blobName, String containerName) {
+
+        var sourceBlob = storageClient.getBlobContainerClient(containerName).getBlobClient(blobName);
+        var targetBlob = storageClient.getBlobContainerClient(containerName + REJECTED_CONTAINER_SUFFIX).getBlobClient(
+            blobName);
+
+        if (targetBlob.exists()) {
+            targetBlob.createSnapshot();
+        }
```

Comment:
perhaps we should create snapshot right before copying the blob?

Reply:
Good point!
Just moving existing code to a new class in this PR.
I'll fix this in a separate PR (and add a test - no need to create a snapshot if file in source container doesn't exist).